tripy/tripy/frontend/trace/ops/slice.py

```diff
@@ -37,6 +37,11 @@ def infer_rank(self):
         # How can we compute the output rank in the case when start, size, stride tensors are dynamic?
         self.outputs[0].rank = self.inputs[0].rank
 
+    def infer_len(self):
+        # Skipping inference here since it depends on the concrete _values_ of the inputs
+        # rather than their shapes. This can be revisited if necessary
+        return [None]
```

Comment:
Not sure we can have dynamic slice on a shape tensor. Even if we could, the shape tensor wouldn't be usable to create an execution tensor since dynamic rank is not supported. Maybe we can assume that the slice bounds are always constants for shape tensors - this will probably be a bit complicated as we would need constant folding to make it work reliably.

Reply:
`tp.Shape`s in Tripy are just a wrapper on ordinary tensors; they are not implemented as TRT shape tensors. Hence, any operation possible on other tensors is possible on them.

That said, we could do inference in this case by having special handling for constants as in some of the other cases in this PR.

Reply:
Added in a workaround for slice, though it's not the prettiest (lots of corner cases, but that is what makes Python slicing semantics user-friendly).

Reply:
Resolving since the new implementation handles the case of a constant slice and it's indeed very likely to be the most common case.